### ISSUE TITLE
How to enhance the monocular branch using the relative depth of Depth_AnythingV2?

### ISSUE THREAD

**Issue by AstraMac** [RM-Zhang/SGCDet#7]
Hello, I have noticed that your supplementary experiment adopts the relative depth derived from Depth_AnythingV2 as an additional input to the monocular branch. Could you please share the specific implementation method?

**Comment by RM-Zhang** [comment]
Hello, thanks for your interest about our work.

We use the relative depth from DAv2 as input to DepthNet's monocular branch.

**Comment by AstraMac** [comment]
Thank you for your previous clarification. It was very helpful.
I noticed that in your codebase, the monocular branch (fnet_mono) takes the backbone features (xs) as input, while the MVS branch takes raw images.Since fnet_mono operates on the feature level, could you please confirm if the following implementation details match your experiments with Depth Anything V2 (DAV2)?
Fusion Strategy: Do you resize the DAV2 depth map to the resolution of the feature map xs and concatenate them?Depth Encoder: Before concatenation, do you simply append the 1-channel depth map (making the input 256+1), or do you use a lightweight encoder (e.g., a few Conv2d layers) to project the depth map into a higher dimension (e.g., 32 channels) for better feature alignment?Normalization: Since DAV2 outputs relative depth with varying scales, did you apply any specific normalization (e.g., Min-Max to [0, 1] or Instance Norm) to the depth values before feeding them into the network?

**Comment by RM-Zhang** [comment]
Please refer to the code below, which I hope will be helpful in answering your question.

```
### init
self.fnet_mono = ConvBnReLU2D(in_channels=1, out_channels=128)
self.mono_regulation = SimpleUnet2D(in_channel=128)

### normalization
def min_max_normalize_per_batch(x: torch.Tensor, eps=1e-6):
    # x: [B, 1, H, W]
    B = x.shape[0]
    x_reshaped = x.view(B, -1)  # [B, H*W]

    x_min = x_reshaped.min(dim=1)[0].view(B, 1, 1, 1)  # [B,1,1,1]
    x_max = x_reshaped.max(dim=1)[0].view(B, 1, 1, 1)

    x_norm = (x - x_min) / (x_max - x_min + eps)
    return x_norm

### forward
with torch.no_grad():
    mono_depth, _ = self.depth_anything.infer(img) # [B, 1, H, W]
    mono_depth = min_max_normalize_per_batch(F.interpolate(mono_depth, scale_factor=(1/stride, 1/stride), mode="bilinear"))
f_mono = self.fnet_mono(mono_depth)
mono_reg = self.mono_regulation(f_mono)
```

**Comment by AstraMac** [comment]
Thank you so much for your detailed reply, it’s really helpful!



---- Replied Message ----
| From | ***@***.***> |
| Date | 01/22/2026 22:16 |
| To | ***@***.***> |
| Cc | ***@***.***>***@***.***> |
| Subject | Re: [RM-Zhang/SGCDet] How to enhance the monocular branch using the relative depth of Depth_AnythingV2? (Issue #7) |
RM-Zhang left a comment (RM-Zhang/SGCDet#7)

Please refer to the code below, which I hope will be helpful in answering your question.

### init
self.fnet_mono = ConvBnReLU2D(in_channels=1, out_channels=128)
self.mono_regulation = SimpleUnet2D(in_channel=128)

### normalization
def min_max_normalize_per_batch(x: torch.Tensor, eps=1e-6):
    # x: [B, 1, H, W]
    B = x.shape[0]
    x_reshaped = x.view(B, -1)  # [B, H*W]

    x_min = x_reshaped.min(dim=1)[0].view(B, 1, 1, 1)  # [B,1,1,1]
    x_max = x_reshaped.max(dim=1)[0].view(B, 1, 1, 1)

    x_norm = (x - x_min) / (x_max - x_min + eps)
    return x_norm

### forward
with torch.no_grad():
    mono_depth, _ = self.depth_anything.infer(img) # [B, 1, H, W]
    mono_depth = min_max_normalize_per_batch(F.interpolate(mono_depth, scale_factor=(1/stride, 1/stride), mode="bilinear"))
f_mono = self.fnet_mono(mono_depth)
mono_reg = self.mono_regulation(f_mono)
